Bottom: 84e1419b715a9de361d3094e9a74db4147c29ded
Top:    0928e91c8abf8b8c0c9ad777bf7cf68c45fef9cf
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-12-24 11:50:24 +0000

Refresh of remove-draw-call-from-resetfou

---

diff --git a/src/action.c b/src/action.c
index c4c28c0..d8e631f 100644
--- a/src/action.c
+++ b/src/action.c
@@ -2365,17 +2365,26 @@ ActionConnection (int argc, char **argv, int x, int y)
 
 	case F_ResetLinesAndPolygons:
 	  if (ResetFoundLinesAndPolygons (true))
-	    IncrementUndoSerialNumber ();
+	    {
+	      IncrementUndoSerialNumber ();
+	      Draw ();
+	    }
 	  break;
 
 	case F_ResetPinsViasAndPads:
 	  if (ResetFoundPinsViasAndPads (true))
-	    IncrementUndoSerialNumber ();
+	    {
+	      IncrementUndoSerialNumber ();
+	      Draw ();
+	    }
 	  break;
 
 	case F_Reset:
 	  if (ResetConnections (true))
-	    IncrementUndoSerialNumber ();
+	    {
+	      IncrementUndoSerialNumber ();
+	      Draw ();
+	    }
 	  break;
 	}
       RestoreCrosshair (true);
@@ -2789,7 +2798,10 @@ ActionDisplay (int argc, char **argv, int childX, int childY)
 	  if (TEST_FLAG (AUTODRCFLAG, PCB) && Settings.Mode == LINE_MODE)
 	    {
 	      if (ResetConnections (true))
-		IncrementUndoSerialNumber ();
+		{
+		  IncrementUndoSerialNumber ();
+		  Draw ();
+		}
 	      if (Crosshair.AttachedLine.State != STATE_FIRST)
 		LookupConnection (Crosshair.AttachedLine.Point1.X,
 				  Crosshair.AttachedLine.Point1.Y, true, 1,
diff --git a/src/find.c b/src/find.c
index 345e670..748431e 100644
--- a/src/find.c
+++ b/src/find.c
@@ -3486,11 +3486,7 @@ ResetFoundPinsViasAndPads (bool AndDraw)
   }
   END_LOOP;
   if (change)
-    {
-      SetChangedFlag (true);
-      if (AndDraw)
-        Draw ();
-    }
+    SetChangedFlag (true);
   return change;
 }
 
@@ -3555,11 +3551,7 @@ ResetFoundLinesAndPolygons (bool AndDraw)
   }
   ENDALL_LOOP;
   if (change)
-    {
-      SetChangedFlag (true);
-      if (AndDraw)
-        Draw ();
-    }
+    SetChangedFlag (true);
   return change;
 }
 
@@ -3922,7 +3914,10 @@ DRCAll (void)
   TheFlag = FOUNDFLAG | DRCFLAG | SELECTEDFLAG;
 
   if (ResetConnections (true))
-    IncrementUndoSerialNumber ();
+    {
+      IncrementUndoSerialNumber ();
+      Draw (); /* XXX: Not sure if this is required */
+    }
 
   User = false;
 
diff --git a/src/report.c b/src/report.c
index 07ff6b8..f0668d9 100644
--- a/src/report.c
+++ b/src/report.c
@@ -46,6 +46,7 @@
 #include "macro.h"
 #include "undo.h"
 #include "find.h"
+#include "draw.h"
 
 #ifdef HAVE_LIBDMALLOC
 #include <dmalloc.h>
@@ -688,7 +689,8 @@ ReportAllNetLengths (int argc, char **argv, int x, int y)
       continue;
 
     got_one:
-      ResetConnections (true);
+      if (ResetConnections (true))
+        Draw ();
       /* NB: XYtoNetLength calls LookupConnection, which performs an undo
        *     serial number update, so we don't need to add one here.
        */
@@ -706,7 +708,8 @@ ReportNetLength (int argc, char **argv, int x, int y)
   char *netname = 0;
   int found = 0;
 
-  ResetConnections (true);
+  if (ResetConnections (true))
+    Draw ();
   /* NB: XYtoNetLength calls LookupConnection, which performs an undo
    *     serial number update, so we don't need to add one here.
    */
diff --git a/src/set.c b/src/set.c
index c5d34eb..25267d8 100644
--- a/src/set.c
+++ b/src/set.c
@@ -293,7 +293,10 @@ SetMode (int Mode)
       if (Mode == LINE_MODE && TEST_FLAG (AUTODRCFLAG, PCB))
 	{
 	  if (ResetConnections (true))
-	    IncrementUndoSerialNumber ();
+	    {
+	      IncrementUndoSerialNumber ();
+	      Draw ();
+	    }
 	}
     }
